baremetal/metal3machine_manager_test.go

```diff
@@ -3472,38 +3437,84 @@ var _ = Describe("Metal3Machine manager", func() {
 	DescribeTable("Test GetMachineDeploymentName",
 		func(tc testCaseGetMachineDeploymentName) {
 			objects := []client.Object{}
-			if tc.expectedMachineSet != nil {
-				objects = append(objects, tc.expectedMachineSet)
+			if tc.Machine != nil {
+				objects = append(objects, tc.Machine)
 			}
-			for _, ms := range tc.MachineSets {
-				objects = append(objects, ms)
+			if tc.MachineSetList != nil {
+				for i := range tc.MachineSetList.Items {
+					objects = append(objects, &tc.MachineSetList.DeepCopy().Items[i])
+				}
 			}
 			c := fakeclient.NewClientBuilder().WithScheme(setupSchemeMm()).WithObjects(objects...).Build()
 			machineMgr, err := NewMachineSetManager(c, tc.Machine,
-				tc.MachineSets, tc.expectedMachineSet, logr.Discard(),
+				tc.MachineSetList, logr.Discard(),
 			)
 			Expect(err).NotTo(HaveOccurred())
 
 			result, err := machineMgr.getMachineDeploymentName(context.TODO())
 			if tc.expectError {
 				Expect(err).To(HaveOccurred())
+				Expect(tc.expectedMachineSet).To(BeNil())
 			} else {
 				Expect(err).NotTo(HaveOccurred())
 			}
-			machineSetObjects := capi.MachineSetList{}
-			for ms := range machineSetObjects.Items {
-				tc.expectedMachineSet = &machineSetObjects.Items[ms]
-				Expect(result).To(Equal(tc.expectedMachineSet))
+			for ms := range tc.MachineSetList.Items {
+				foundMS := &tc.MachineSetList.Items[ms]
+				key := client.ObjectKey{
+					Name:      foundMS.Name,
+					Namespace: foundMS.Namespace,
+				}
+				err = c.Get(context.TODO(), key, foundMS)
```

Comment:
I'm not sure we're actually checking anything here? We already know that the fake client is loaded with these objects because they were added in line 3443-3448 above.

Reply:
@macaptain that is to check if can get a MachineSet I want in the same namespace and with the name. Lines 3443-3448 above are adding items to the MachineSetList.

Reply:
Lines 3443-3448 add the MachineSetList Items specified in the test case to the fakeClient:

https://github.com/metal3-io/cluster-api-provider-metal3/blob/e1cb70ff290868eb8913d37cfa947da14fa57a5d/baremetal/metal3machine_manager_test.go#L3444-L3445

Then we're looping over the same list `tc.MachineSetList.Items` and checking we can fetch them out of the fakeClient with `c.Get`. But we know we can: we put them there, so I don't see how the `err` can ever be hit. We're looping over the same list twice.

I am not convinced these lines can ever fail, no matter what is in the test case, but not a blocker for me, just an observation, the tests themselves are still good. Let me know what you think!

Reply:
I have checked it a bit closer, and indeed, I have overlooked it. Initially, when started writing the tests, the structure of them was a bit different and I was expecting  `foundMS` to be passed to `Expect(result).To(Equal(tc.expectedMDName))`, but that is not the case now, I have changed them later, so that was left out and I removed it here and on the next tests. Thanks for pointing it out.